### PR TITLE
Fix the slow integration tests.

### PR DESCRIPTION
**Pull request by bemasc**
It seems that these tests have been broken since the introduction
of CHURN, at least.

Fixes https://github.com/uProxy/uproxy/issues/1849

---

This change is [Reviewable](https://reviewable.io/reviews/uproxy/uproxy-lib/384)

### CODE REVIEW

File: src/integration-tests/socks-echo/slow.core-env.spec.ts [hunk 1]
```diff
@@ -49,26 +49,34 @@ function slowTestDescription(useChurn:boolean) {
...
+        let id = event.connectionId
+        if (id in counters) {
+          counters[id] += event.response.byteLength;
+        } else {
+          throw new Error('Unexpected connectionId ' + id);
+        }
+        if (counters[id] === repeat * blockSize) {
+          delete counters[id];
+          // Check if we have deleted the last id.
+          let empty = true;
```
Comment (trevj) [inline src/integration-tests/socks-echo/slow.core-env.spec.ts]: How about:
```typescript
if (Object.keys(counters) === 0) {
  done();
}
```

File: src/integration-tests/socks-echo/slow.core-env.spec.ts [hunk 2]
```diff
@@ -49,26 +49,34 @@ function slowTestDescription(useChurn:boolean) {
...
-        var outputString = '';
-        testModule.on('receivedData', (event:ReceivedDataEvent) => {
-          if (event.connectionId != connectionId) {
-            return;
+      // Maps connectionIds to the number of bytes received so far.  Counters
+      // start at 0, and entries are deleted when all data has been received
+      // for a connection.  When all entries have been deleted, the test passes.
+      let counters :{[id:string]: number} = {};
+      testModule.on('receivedData', (event:ReceivedDataEvent) => {
+        let id = event.connectionId
```
Comment (trevj) [inline src/integration-tests/socks-echo/slow.core-env.spec.ts]: missing semi-colon (also, this count be `const`)

**Comment by trevj** [comment]
Fantastic!

File: src/integration-tests/socks-echo/slow.core-env.spec.ts (hunk 1 above)
Comment (bemasc) [inline src/integration-tests/socks-echo/slow.core-env.spec.ts]: Done

File: src/integration-tests/socks-echo/slow.core-env.spec.ts (hunk 2 above)
Comment (bemasc) [inline src/integration-tests/socks-echo/slow.core-env.spec.ts]: Done.

**Pull request merged**